rust_src/src/fonts.rs

```diff
@@ -112,4 +113,13 @@ pub fn fontp(object: LispObject, extra_type: LispObject) -> bool {
     })
 }
 
+/// Return t if and only if font-spec SPEC matches with FONT.  FONT is a font-spec, font-entity,
+/// or font-object.
+#[lisp_fn]
+pub fn font_match_p(spec: LispObject, font: LispObject) -> bool {
+    if !spec.is_font_spec() {wrong_type!(Qfont_spec, spec)}
```

Comment:
Is there a better pattern for here? I held off implementing anything bigger since I saw similar patterns elsewhere, but I seem to remember some kind of From impl for Option letting me automagically use types in my function signatures?